core/src/main/java/io/grpc/inprocess/InProcessChannelBuilder.java

```diff
@@ -61,14 +62,22 @@ private InProcessChannelBuilder(String name) {
   }
 
   @Override
-  protected ChannelEssentials buildEssentials() {
-    final String name = this.name;
-    ClientTransportFactory transportFactory = new ClientTransportFactory() {
-      @Override
-      public ClientTransport newClientTransport() {
-        return new InProcessTransport(name);
-      }
-    };
-    return new ChannelEssentials(transportFactory, null);
+  protected ClientTransportFactory buildTransportFactory() {
+    return new InProcessClientTransportFactory();
+  }
+
+  private final class InProcessClientTransportFactory extends AbstractReferenceCounted
+          implements ClientTransportFactory {
+    private final String name = InProcessChannelBuilder.this.name;
```

Comment:
Why have this field? I'd understand if it made this class static, but it doesn't.